druid-shell/Cargo.toml

```diff
@@ -100,14 +100,14 @@ bitflags = "1.3.2"
 [target.'cfg(any(target_os = "freebsd", target_os="linux", target_os="openbsd"))'.dependencies]
 ashpd = { version = "0.3.2", optional = true }
 # TODO(x11/dependencies): only use feature "xcb" if using X11
-cairo-rs = { version = "0.16.7", default-features = false, features = ["xcb"] }
-cairo-sys-rs = { version = "0.16.3", default-features = false, optional = true }
+cairo-rs = { version = "0.18.0", default-features = false, features = ["xcb"] }
+cairo-sys-rs = { version = "0.18.0", default-features = false, optional = true }
 futures = { version = "0.3.26", optional = true, features = ["executor"]}
-gdk-sys = { version = "0.16.0", optional = true }
+gdk-sys = { version = "0.18.0", optional = true }
 # `gtk` gets renamed to `gtk-rs` so that we can use `gtk` as the feature name.
-gtk-rs = { version = "0.16.2", package = "gtk", optional = true }
-glib-sys = { version = "0.16.3", optional = true }
-gtk-sys = { version = "0.16.0", optional = true }
+gtk-rs = { version = "0.18.0", package = "gtk", optional = true }
```

Comment:
from gtk, you can go to `gtk::gdk::ffi` for example, all those dependencies are not needed. You could just depend on `gtk` except for cairo where you enable `xcb` feature